File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/UserDao.java
package seigneur.gauvain.mycourt.data.local.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import io.reactivex.Completable;
import io.reactivex.Maybe;
import io.reactivex.Single;
import seigneur.gauvain.mycourt.data.model.User;

@Dao
public interface UserDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertUser(User user);

    @Query("SELECT * FROM user")
    Maybe<User> getUser();

    @Query("SELECT * FROM user WHERE isAllowedToUpload")
    boolean checkIfUserIsAllowedToUpload();

    @Delete
    void deleteUser(User user);

}

File: app/src/main/java/seigneur/gauvain/mycourt/data/local/dao/UserDao.kt
package seigneur.gauvain.mycourt.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Maybe
import seigneur.gauvain.mycourt.data.model.User

@Dao
interface UserDao {

    @get:Query("SELECT * FROM user")
    val user: Maybe<User>

    //for testing
    @get:Query("SELECT * FROM user")
    val userLive: LiveData<User>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(user: User): Long

    @Query("SELECT * FROM user WHERE isAllowedToUpload")
    fun checkIfUserIsAllowedToUpload(): Boolean

    @Delete
    fun deleteUser(user: User)

}
